src/makeInfinityAdder.js

```diff
@@ -5,6 +5,23 @@
  */
 function makeInfinityAdder() {
   // write code here
+  let sum = 0;
+
+  const sumNumber = (number) => {
+    if (number === undefined) {
+      const result = sum;
+
+      sum = 0;
+
+      return result;
+    }
+
+    sum += number;
```

Comment:
The addition should be done in a new function call to support the chaining as required by the task.